### ISSUE TITLE
Controlled miner mining slowly due to randomness

### ISSUE THREAD

**Issue by Chrysaloid** [PrinceOfAmber/Cyclic#2151]
Minecraft Version: 1.12.2
Forge Version: forge-1.12.2-14.23.5.2859
Mod Version: Cyclic-1.12.2-1.20.12
Single Player or Server: Single Player

Describe problem (what you were doing; what happened; what should have happened):

Why is Controlled miner mining random blocks in its operating area? Due to this if I make it mine large area the last blocks take ages to get mined. Is this expected behaviour? And if yes then why? Why can't it work just like forester or like structure builder so checking the area layer by layer and line by line? I'm very frustrated because of this. I have to build multiple controlled miners to achieve desired efficiency of mining and this cost lots of energy and I can't afford that.

Log file link: None needed.
Video/images/gifs (direct upload or link): None needed.

**Comment by Lothrazar** [comment]
Yep, in 1.12 it was randomized and really slow.
in 1.16/1.18+ it follows a pattern and is more straight up like this. Also it can go up or down.  so i would need to backport those behaviors 
![controlled miner quarry mode](https://user-images.githubusercontent.com/621529/200101757-4544fadc-8a3c-4b3d-a724-8115ffdcdaba.gif)

**Comment by Chrysaloid** [comment]
Yeah... Newer Versions... I would switch to them if other mods that I use had working newer versions. At the moment I'm stuck in 1.12.2 due to other modes not receiving updates. **If you would backport this behaviour I (probably I wouldn't be the only one) would be super happy.** If not I'll probably have to try again recreate my modpack in newer version swapping not working mods to other ones with, hopefully the same functionality.
